morf-core/src/main/java/org/alfasoftware/morf/sql/element/FieldLiteral.java

```diff
@@ -239,8 +366,40 @@ protected AliasedField deepCopyInternal(DeepCopyTransformation transformer) {
    */
   @Override
   public FieldLiteral as(String aliasName) {
-    super.as(aliasName);
-    return this;
+    return (FieldLiteral) super.as(aliasName);
+  }
+
+
+  /**
+   * @see java.lang.Object#equals(java.lang.Object)
+   */
+  @Override
+  public boolean equals(Object obj) {

```

Comment:
Elsewhere you're using the Eclipse auto-gen equals (I think), here: EqualsBuilder. Do we want to be consistent?

Reply:
Can do.  Bump this up to a review issue if you like and I'll tidy it up.